src/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,3808 @@
+    #B = Mat(B)
+    res = f1*A - B*g1
+    res = reshape(res, 1, length(res))
+    for i=1:length(res)
+      delta[j,i] = res[1,i]
+    end
+  end
+
+  #Mat = AbstractAlgebra.MatrixSpace(R, size(delta)...)
+  #display(delta)
+  gamma = matrix_kernel(delta)
+
+  t2 = size(g2)[1]
+  n = m
+  m = s0*t1 + s1*t2
+  #rho::Matrix{T} = zeros(R, n, m)
+  rho::MatrixElem{T} = zero_matrix(R, m,n)
+  for j=1:m
+    #b_vector::Vector{T} = [R(0) for i=1:m]
+    b_vector = zero_matrix(R, 1,m)
+    b_vector[1,j] = R(1)
+    C = reshape(b_vector[1,1:s0*t1], s0, t1)
+    #Mat = AbstractAlgebra.MatrixSpace(R, t1, s0)
+    #C = Mat(C)
+    D = reshape(b_vector[1,s0*t1+1:length(b_vector)], s1, t2)
+    #Mat = AbstractAlgebra.MatrixSpace(R, t2, s1)
+    #D = Mat(D)
+    res1 = C*g1
+    res1 = reshape(res1, 1,length(res1))
+    res2 = f1*C - D*g2
+    res2 = reshape(res2, 1,length(res2))
+    for i=1:length(res1)
+        rho[j,i] = res1[1,i]
+    end
+    for i=1:length(res2)
+        #rho[:,j] = vcat(vec(res1), vec(res2))
+        rho[j,i+length(res1)] = res2[1,i]
+    end
+  end
+
+  M = SubQuo(gamma, rho)
+
+  function convert_to_matrix(v::SubQuoElem{T})
+    if parent(v) !== M
+      throw(DomainError("v does not represent a homomorphism"))
+    end
+    R = base_ring(M)
+    #A = AbstractAlgebra.matrix(R,map(x -> convert_poly(R,x),Array(v.singular_v)))
+    #A = map(x -> convert_poly(R,x),Array(v.singular_v))
+    #A = AbstractAlgebra.matrix(R, t0, s0, A[1:s0*t0])'
+    A = reshape(dense_row(v.repres.coords[R, 1:s0*t0], s0*t0), s0, t0)
+    #A = matrix(R, t0, s0, dense_row(v.repres.coords[R, 1:s0*t0], s0*t0))'
+    #A = v.v*v.parent_sq.generators  # ineffizienter
+    #A = reshape(A[1,1:s0*t0], s0, t0)
+    #N = cokernel(g1)
+    #for i=1:s0ems) <: Tuple ? [i for i in elems] : [elems])
+
+    #    row = lift(N,A[i,:])
+    #    for j=1:t0
+    #        A[i,j]=row[1,j]
+    #    end
+    #end
+    return A
+  end
+
+  return M, convert_to_matrix
+end
+
+# lookup-tables only used in the old code
+# lookup-tables to find the correct function, that converts subquotient elements to homomorphisms or homomorphisms to subquotient elements:
+#global subquotient_elem_to_homomorphism_lookup_table = IdDict{Subquotient,Function}()
+#global homomorphism_to_subquotient_elem_lookup_table = IdDict{Tuple{Subquotient,Subquotient},Function}()
+
+
+# since hom yields possibly wrong results, this is an alternative implementation for comparison, 
+# which is known to work correctly in various examples (by comparing with Macaulay2 and by testing 
+# whether the homomorphism corresponding to elements of the subquotient are well-defined
+# the function also supports simplification of the subquotient, while retaining the correspondence to 
+# morphisms, which is essential for checking correctness and practical use of the output
+@doc Markdown.doc"""
+  hom2(M::Subquotient,N::Subquotient)
+> Return a subquotient S such that $Hom(M,N) \cong S$
+"""
```

Comment:
What does this mean? If `hom` is wrong, then we would like to see an example.

Eventually there should be no more `hom2`.

Reply:
Hom and Hom2 implement the same mathematical algorithm, but the implementations differ a lot e.g. in the data structures. As a result performance differs depending on the example favoring the one or the other. So it makes sense to offer both. We plan to combine in one command and make the choice optional, or get rid of the two alternatives if at some point there is consistent performance. 

Independent of this, for some settings where the answer is a free module (in the tests), the result of Hom is not per se free (though isomorphic to a free module), while that of Hom2 is free. We are analyzing this. It might point to a bug in Hom.

We don't mind to update before or after merge.